src/app/3d/qgs3dviewsmanagerdialog.cpp

```diff
@@ -0,0 +1,200 @@
+  }
+
+  QgsProject::instance()->setDirty();
+  currentChanged( m3DViewsListView->selectionModel()->currentIndex(), m3DViewsListView->selectionModel()->currentIndex() );
+}
+
+void Qgs3DViewsManagerDialog::duplicateClicked()
+{
+  if ( m3DViewsListView->selectionModel()->selectedRows().isEmpty() )
+    return;
+
+  QString existingViewName = m3DViewsListView->selectionModel()->selectedRows().at( 0 ).data( Qt::DisplayRole ).toString();
+  QString newViewName = askUserForATitle( existingViewName, tr( "Duplicate" ), false );
+
+  if ( newViewName.isEmpty() )
+    return;
+
+  QgisApp::instance()->duplicate3DMapView( existingViewName, newViewName );
+
+  QgsProject::instance()->setDirty();
+}
+
+void Qgs3DViewsManagerDialog::removeClicked()
+{
+  if ( m3DViewsListView->selectionModel()->selectedRows().isEmpty() )
+    return;
+
+  QString warningTitle = tr( "Remove 3D View" );
+  QString warningMessage = tr( "Do you really want to remove selected 3D view?" );
+
+  if ( QMessageBox::warning( this, warningTitle, warningMessage, QMessageBox::Ok | QMessageBox::Cancel ) != QMessageBox::Ok )
+    return;
+
+  QString viewName = m3DViewsListView->selectionModel()->selectedRows().at( 0 ).data( Qt::DisplayRole ).toString();
+
+  QgsProject::instance()->views3DManager()->remove3DView( viewName );
+  if ( Qgs3DMapCanvasDockWidget *w = QgisApp::instance()->findChild<Qgs3DMapCanvasDockWidget *>( viewName + QStringLiteral( "DockObject" ) ) )
+    w->close();
+
+  QgsProject::instance()->setDirty();
+}
+
+void Qgs3DViewsManagerDialog::renameClicked()
+{
+  if ( m3DViewsListView->selectionModel()->selectedRows().isEmpty() )
+    return;
+
+  QString oldTitle = m3DViewsListView->selectionModel()->selectedRows().at( 0 ).data( Qt::DisplayRole ).toString();
+  QString newTitle = askUserForATitle( oldTitle, tr( "Rename" ), true );
+
+  if ( newTitle.isEmpty() )
+    return;
+
+  QgsProject::instance()->views3DManager()->rename3DView( oldTitle, newTitle );
+
+  if ( Qgs3DMapCanvasDockWidget *widget = QgisApp::instance()->findChild<Qgs3DMapCanvasDockWidget *>( oldTitle + QStringLiteral( "DockObject" ) ) )
+  {
+    widget->setWindowTitle( newTitle );
+    widget->setObjectName( newTitle + QStringLiteral( "DockObject" ) );
+    widget->mapCanvas3D()->setObjectName( newTitle );
+  }
+
+  QgsProject::instance()->setDirty();
+}
+
+void Qgs3DViewsManagerDialog::currentChanged( const QModelIndex &current, const QModelIndex &previous )
+{
+  Q_UNUSED( previous );
+
+  mRenameButton->setEnabled( current.isValid() );
+  mRemoveButton->setEnabled( current.isValid() );
+  mDuplicateButton->setEnabled( current.isValid() );
+  if ( !current.isValid() )
+  {
+    mShowButton->setEnabled( false );
+    mHideButton->setEnabled( false );
+    return;
+  }
+
+  QString viewName = current.data( Qt::DisplayRole ).toString();
+  bool isOpen = QgsProject::instance()->views3DManager()->get3DViewSettings( viewName ).attribute( QStringLiteral( "isOpen" ), QStringLiteral( "0" ) ).toInt() == 1;
```

Comment:
this is messy -- we should keep the internal dom manipulation/attribute retrieval centralized in the view manager class only. (Otherwise this is highly likely to get out-of-sync with changes made in that class.)

Better though would be to keep QgisApp::instance() as responsible for tracking the open 3d dock widgets, and then here we could just call something like `QgisApp::instance()->is3DViewOpen( viewName )`. We could do this by adding a `QMap< QString, QPointer< Qgs3DMapCanvasDockWidget > > mOpen3DDocks` to QgisApp. This would also allow us to replace the `QgisApp::instance()->findChild<Qgs3DMapCanvasDockWidget *>( title + QStringLiteral( "DockObject" ) ) )` calls elsewhere here too, which also look a bit fragile.

Reply:
Insetead of a Map I added a QSet to record open 3D views because by having a map we will need to add a rename function..